app/src/main/java/com/sun_asterisk/foodies/screen/main/MainActivity.kt

```diff
@@ -2,12 +2,46 @@ package com.sun_asterisk.foodies.screen.main
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.viewpager.widget.ViewPager
 import com.sun_asterisk.foodies.R
+import com.sun_asterisk.foodies.screen.favorite.FavoriteFragment
+import com.sun_asterisk.foodies.screen.home.HomeFragment
+import com.sun_asterisk.foodies.screen.main.layout_adapter.ViewPagerAdapter
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        setupTabLayout()
+    }
+
+    private fun setupViewPager(viewPager: ViewPager) {
+        viewPager.adapter = ViewPagerAdapter(supportFragmentManager).apply {
+            addFragmentContainer(HomeFragment(), resources.getString(R.string.home))
+            addFragmentContainer(FavoriteFragment(), resources.getString(R.string.favorite))
+        }
+    }
+
+    private fun setupTabIcons() {
+        val icons = resources.obtainTypedArray(R.array.layout_icons)
+        tabLayout.getTabAt(MainPage.HOME_PAGE)
+            .let { it?.setIcon(icons.getDrawable(MainPage.HOME_PAGE)) }
+        tabLayout.getTabAt(MainPage.FAVORITE_PAGE)
+            .let { it?.setIcon(icons.getDrawable(MainPage.FAVORITE_PAGE)) }
+    }
+
+    private fun setupTabLayout() {
+        setupViewPager(viewPager)
+        tabLayout.setupWithViewPager(viewPager)
+        setupTabIcons()
+    }
+
+    companion object MainPage {
+        const val HOME_PAGE = 0
+        const val FAVORITE_PAGE = 1
```

Comment:
Để private có được không ? 
```suggestion
    companion object {
        private const val HOME_PAGE = 0
        const val FAVORITE_PAGE = 1
```